jadx-gui/src/main/java/jadx/gui/device/protocol/ADB.java

```diff
@@ -148,17 +135,18 @@ public static boolean startServer(String adbPath, int port) throws IOException {
 			proc.destroyForcibly();
 			return false;
 		}
-		InputStream is = proc.getInputStream();
-		int size = is.available();
-		byte[] bytes = new byte[size];
-		is.read(bytes, 0, size);
-		return new String(bytes).contains(tcpPort);
+		ByteArrayOutputStream out = new ByteArrayOutputStream();
+		try (InputStream in = proc.getInputStream()) {
+			int read;
+			byte[] buf = new byte[1024];
+			while ((read = in.read(buf)) >= 0) {
+			}
+		}
+		return new String(out.toByteArray()).contains(tcpPort);
```

Comment:
Write to `out` variable is missing here

Reply:
sorry stupid bug. Seems like I am getting laze as usually I use `InputStream.getAllBytes()`